docs/extending/editor_api.md

```diff
@@ -0,0 +1,168 @@
+# Accessing the editor programmatically
+
+Wagtail's editor is built with various JavaScript components that can be interacted with programmatically. This document provides an overview of how to access and extend the editor's functionality.
+
+## The editor's `<form>` element
+
+The editor's main `<form>` element can be queried using the `data-edit-form` attribute. This is useful for attaching event listeners or manipulating the form programmatically, as well as getting the form's `FormData` representation.
```

Comment:
This isn't used in the example because we attach the controller directly via the `FieldPanel`, so we're already in the context of the `<form>`. Not sure if I can think of other examples to add to the docs that utilise this